src/Symfony/Component/Console/Command/Command.php

```diff
@@ -420,14 +418,14 @@ public function getNativeDefinition()
     /**
      * Adds an argument.
      *
-     * @param int|null             $mode    The argument mode: InputArgument::REQUIRED or InputArgument::OPTIONAL
-     * @param string|string[]|null $default The default value (for InputArgument::OPTIONAL mode only)
+     * @param $mode    The argument mode: InputArgument::REQUIRED or InputArgument::OPTIONAL
+     * @param $default The default value (for InputArgument::OPTIONAL mode only)
      *
      * @throws InvalidArgumentException When argument mode is not valid
      *
      * @return $this
      */
-    public function addArgument(string $name, int $mode = null, string $description = '', $default = null)
+    public function addArgument(string $name, int $mode = null, string $description = '', mixed $default = null)
```

Comment:
Was the parameter annotation wrong here?

```suggestion
    public function addArgument(string $name, int $mode = null, string $description = '', string|array|null $default = null)
```

Reply:
Tests fail if we restrict here, we should at least allow `int|float|bool`. But I don't see the need to restrict when it was allowed before. We should avoid needless BC breaks IMHO.

Reply:
It does not make much sense to pass a default value that cannot be produced by invoking the command via the console. `null` would be the only reasonable exception here.

If we don't want to break BC, I'm fine with `mixed` but please let's keep the more precise type in the doc block then.

Reply:
What would be the point of the annotation? if technically we don't care, we shouldn't bother our users IMHO.
I'm sure there are ppl out there that already found some creative use of that :)

Reply:
> I'm sure there are ppl out there that already found some creative use of that :)

That's precisely what scares me. 🙈

Reply:
We should be proud of what ppl do with Symfony :)

Reply:
I did some `git blame` + some code inspection. We use mixed in many places already. Having this restricted list is both wrong (as proved by tests) and not needed. I confirm this change.

PR ready.